[BUG] Array index out of bounds when running iceberg query

**Describe the bug**

User faced following error when running iceberg query against tpch:

Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at com.nvidia.spark.rapids.iceberg.parquet.GpuIcebergParquetReader.$anonfun$filterParquetBlocks$3(reader.scala:206)
	at com.nvidia.spark.rapids.iceberg.parquet.GpuIcebergParquetReader.$anonfun$filterParquetBlocks$3$adapted(reader.scala:206)

**Steps/Code to reproduce bug**

Run against tpch dataset.

**Expected behavior**

Should not throw error.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/parquet/reader.scala
+++ b/iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/parquet/reader.scala
@@ -175,17 +175,9 @@ trait GpuIcebergParquetReader extends Iterator[ColumnarBatch] with AutoCloseable
         conf.caseSensitive)
 
       blocks.filter { case (rowGroup, _) =>
-          val beforeOrdinal = rowGroup.getOrdinal
-          val filterResult = statsFilter.shouldRead(typeWithIds, rowGroup) &&
+          statsFilter.shouldRead(typeWithIds, rowGroup) &&
             dictFilter.shouldRead(typeWithIds, rowGroup, reader.getDictionaryReader(rowGroup)) &&
             bloomFilter.shouldRead(typeWithIds, rowGroup, reader.getBloomFilterDataReader(rowGroup))
-
-          val afterOrdinal = rowGroup.getOrdinal
-          if (beforeOrdinal != afterOrdinal) {
-            throw new IllegalStateException(s"Parquet row group ordinal changed from " +
-              s"$beforeOrdinal to $afterOrdinal after filter, this should never happen!")
-          }
-          filterResult
       }
     }.getOrElse(blocks)
       .toSeq

